mlrun/api/db/sqldb/db.py

```diff
@@ -2275,6 +2279,28 @@ def _filter_artifacts_by_kinds(
                 filtered_artifacts.append(artifact)
         return filtered_artifacts
 
+    # TODO - this is a hack needed since link artifacts will be returned even for artifacts of
+    #        the wrong category. Remove this when we refactor this area.
+    def _filter_out_extra_link_artifacts(self, session, artifacts, category):
+        expected_kinds, exclude = category.to_kinds_filter()
+        filtered_artifacts = []
+        for artifact in artifacts:
+            artifact_kind = artifact.struct.get("kind")
+            if artifact_kind != "link":
+                filtered_artifacts.append(artifact)
+                continue
+            linked_artifact = self.read_artifact(
+                session,
+                key=artifact.key,
+                iter=artifact.struct.get("link_iteration"),
+                project=artifact.project,
+            )
```

Comment:
When listing artifacts with category filter - we're going to do read operation for every link artifact in the DB - I think this is too heavy - we need to find something better

Reply:
As discussed - switched to a different solution.